semantic_release/repository.py

```diff
@@ -0,0 +1,62 @@
+"""Helper for using Twine to upload to an artifact repository.
+"""
+import logging
+import os
+from typing import List
+
+from invoke import run
+
+from semantic_release import ImproperConfigurationError
+
+from semantic_release.settings import config
+
+from .helpers import LoggedFunction
+
+logger = logging.getLogger(__name__)
+
+
+class Repository(object):
```

Comment:
What is the benefit of turning the function that was here into a class?

Reply:
As stated in the PR, I'm not the original author of most of the code, so I can just take a guess. The original author is Benjamin René from #303 
My contributions are documentation, cleanups & fixing the failed tests.

Reply:
Oh, sorry 🙈

Reply:
No worries, I made some progress last weekend but still needs some cleaning up before I can put it up. At least it's working now, but still far from perfect.
Still I just pushed it to get some feedback.